components/canvas/canvas_paint_thread.rs

```diff
@@ -683,9 +683,10 @@ impl<'a> CanvasPaintThread<'a> {
         let draw_target = self.drawtarget.create_similar_draw_target(&Size2D::new(source_rect.size.width as i32,
                                                                                   source_rect.size.height as i32),
                                                                      self.drawtarget.get_format());
-        let matrix = Matrix2D::identity().translate(-source_rect.origin.x as AzFloat,
-                                                    -source_rect.origin.y as AzFloat)
-                                         .mul(&self.state.transform);
+        let matrix = Matrix2D::identity()
+            .post_translated(-source_rect.origin.x as AzFloat,
+                             -source_rect.origin.y as AzFloat)
+            .post_mul(&self.state.transform);
```

Comment:
This should be pre_mul (and strictly speaking pre_translated above, but it doesn't matter since it is applied to identity). Or .post_mul(..).post_translate(..)